### ISSUE TITLE
Type Error in MrDMD.fit 

### ISSUE THREAD

**Issue by Reyna-RT** [mathLab/PyDMD#76]
Hi, I started using the MrDMD class, but if I pass a real matrix to the function I get the next error:

`TypeError:Cannot cast ufunc subtract output from dtype('complex128') to dtype('float64') with casting rule 'same_kind'`

It seems to come form the fit function in line 288:

`Xraw -= modes.dot(Psi)`

I fixed it momentarily by adding imaginary zeros to my input matrix data, but it will be nice to let it be real.

**Comment by ndem0** [comment]
Thank you very much for the segnalation.
As soon as possible we will fix the issue.

**Comment by gwb84** [comment]
I get the error too. My input data is real-valued data imported from a .mat file with the hdf5storage package. It is dtype '<f8'. I am using Python 3.5.5 in Anaconda3 on Win7. 

I changed the dtype to 'complex128' and MrDMD.fit worked:

`snapshots = [np.array(snapshots[t],dtype='complex128') for t in range(0,len(snapshots))]`

**Comment by Reyna-RT** [comment]
Sorry for the delay I am using Mac OS High Sierra and Python 2.7.14

**Comment by ndem0** [comment]
Sorry for the late answer.
I tried to reproduce the error using MrDMD on complex matrix, but I didn't get the TypeError.
Please can you try to run the code below and tell me if you get any warning? Thanks a lot for the collaboration.

```
from pydmd import MrDMD
import numpy as np

data = np.random.rand(400, 20).astype(np.complex)
dmd = MrDMD()
dmd.fit(data)
```

**Comment by Reyna-RT** [comment]
Hi, I get a warning that says:

`/Users/reynar/anaconda2/lib/python2.7/site-packages/pydmd/dmdbase.py:341: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
  a = np.linalg.lstsq(modes, snapshots.T[0])[0]`

**Comment by luckystarufo** [comment]
I guess the problem isn't about 'using MrDMD on complex matrix', it's about 'using MrDMD on real matrix'. The computation makes the 'modes.dot(Psi)' complex in line 288 of mrDMD, however, Xraw is real. This leads to the error:

TypeError:Cannot cast ufunc subtract output from dtype('complex128') to dtype('float64') with casting rule 'same_kind'

**Comment by Reyna-RT** [comment]
Yes, it is when I use real data.

**Comment by ndem0** [comment]
Bug fixed in #77 
The fixed version 0.2.1 has be uploaded on PyPi.

**Comment by Reyna-RT** [comment]
Excellent! Thank you!
